plugins/repository-s3/src/main/java/org/elasticsearch/cloud/aws/AwsS3Service.java

```diff
@@ -19,151 +19,156 @@
     /**
      * cloud.aws.proxy.username: In case of proxy with auth, define the username. Shared with discovery-ec2 plugin
      */
-    SecureSetting<SecureString> PROXY_USERNAME_SETTING =
-        SecureSetting.secureString("cloud.aws.proxy.username", null, true, Property.Shared);
-
+    Setting<SecureString> LEGACY_PROXY_USERNAME_SETTING = new Setting<>("cloud.aws.proxy.username", "", SecureString::new,
+        Property.NodeScope, Property.Deprecated, Property.Shared);
     /**
      * cloud.aws.proxy.password: In case of proxy with auth, define the password. Shared with discovery-ec2 plugin
      */
-    SecureSetting<SecureString> PROXY_PASSWORD_SETTING =
-        SecureSetting.secureString("cloud.aws.proxy.password", null, true, Property.Shared);
+    Setting<SecureString> LEGACY_PROXY_PASSWORD_SETTING = new Setting<>("cloud.aws.proxy.password", "", SecureString::new,
+        Property.NodeScope, Property.Filtered, Property.Deprecated, Property.Shared);
     /**
      * cloud.aws.signer: If you are using an old AWS API version, you can define a Signer. Shared with discovery-ec2 plugin
      */
-    Setting<String> SIGNER_SETTING = Setting.simpleString("cloud.aws.signer", Property.NodeScope, Property.Shared);
+    Setting<String> LEGACY_SIGNER_SETTING = Setting.simpleString("cloud.aws.signer",
+        Property.NodeScope, Property.Deprecated, Property.Shared);
     /**
      * cloud.aws.region: Region. Shared with discovery-ec2 plugin
      */
-    Setting<String> REGION_SETTING =
-        new Setting<>("cloud.aws.region", "", s -> s.toLowerCase(Locale.ROOT), Property.NodeScope, Property.Shared);
+    Setting<String> LEGACY_REGION_SETTING = new Setting<>("cloud.aws.region", "", s -> s.toLowerCase(Locale.ROOT),
+        Property.NodeScope, Property.Deprecated, Property.Shared);
     /**
      * cloud.aws.read_timeout: Socket read timeout. Shared with discovery-ec2 plugin
      */
-    Setting<TimeValue> READ_TIMEOUT = Setting.timeSetting("cloud.aws.read_timeout",
-        TimeValue.timeValueMillis(ClientConfiguration.DEFAULT_SOCKET_TIMEOUT), Property.NodeScope, Property.Shared);
+    Setting<TimeValue> LEGACY_READ_TIMEOUT = Setting.timeSetting("cloud.aws.read_timeout",
+        TimeValue.timeValueMillis(ClientConfiguration.DEFAULT_SOCKET_TIMEOUT), Property.NodeScope, Property.Deprecated, Property.Shared);
 
     /**
      * Defines specific s3 settings starting with cloud.aws.s3.
      */
     interface CLOUD_S3 {
         /**
          * cloud.aws.s3.access_key: AWS Access key specific for S3 API calls. Defaults to cloud.aws.access_key.
-         * @see AwsS3Service#KEY_SETTING
+         * @see AwsS3Service#LEGACY_KEY_SETTING
          */
-        SecureSetting<SecureString> KEY_SETTING = SecureSetting.secureString("cloud.aws.s3.access_key", AwsS3Service.KEY_SETTING, true);
+        Setting<SecureString> KEY_SETTING =
+            new Setting<>("cloud.aws.s3.access_key", AwsS3Service.LEGACY_KEY_SETTING, SecureString::new,
```

Comment:
While we are at it, I'd really like to move our `repository-s3` settings under a unique namespace `repositories.s3` instead.

`cloud.aws` prefix was nice when we were sharing same credentials and when we are using `cloud.enabled: true` but I'm sure we don't need that namespace anymore.

Can you do that in the same PR or open a new issue so we track it?

Same for package names, we should move all what we can under a unique package name. So debugging will be much easier (I mean setting log levels on a unique package instead of many).